mangaki/mangaki/admin.py

```diff
@@ -8,13 +8,126 @@
+def merge_works(request, works_to_merge, admin_queryset, from_cluster):
+    works_to_merge = works_to_merge.order_by('id')
+    if not from_cluster:  # I merge those works myself as an admin
+        cluster = WorkCluster(user=request.user, checker=request.user)
+        cluster.save()
+        for work in works_to_merge:
+            cluster.works.add(work)
+    else:  # I review an existing cluster
+        cluster = admin_queryset.first()
+        cluster.checker = request.user
+    if request.POST.get('confirm'):
+        kept_id = int(request.POST.get('id'))
+        kept_work = Work.objects.get(id=kept_id)
+        for field in request.POST.get('fields_to_choose').split(','):
+            if request.POST.get(field) != 'None':
+                setattr(kept_work, field, request.POST.get(field))
+        for work_to_redirect in works_to_merge:
+            if work_to_redirect.id == kept_id:
+                continue
+            for rating_to_redirect in work_to_redirect.rating_set.all():
+                try:
+                    kept_rating = Rating.objects.get(work_id=kept_id, user_id=rating_to_redirect.user_id)
+                    if kept_rating.date and rating_to_redirect.date and kept_rating.date < rating_to_redirect.date:  # Kept rating is not the latest given
+                        kept_rating.choice = rating_to_redirect.choice  # Update the kept rating
+                    rating_to_redirect.delete()
+                except Rating.DoesNotExist:
+                    rating_to_redirect.work_id = kept_id  # Safely transfer the rating to the kept work
+                    rating_to_redirect.save()
+            for staff_to_redirect in work_to_redirect.staff_set.all():
+                if Staff.objects.filter(work_id=kept_id, artist_id=staff_to_redirect.artist_id, role_id=staff_to_redirect.role_id).exists():  # This staff is already in the kept work
+                    staff_to_redirect.delete()
+                else:
+                    staff_to_redirect.work_id = kept_id  # Safely transfer the staff to the kept work
+                    staff_to_redirect.save()
+            for genre in work_to_redirect.genre.all():
+                kept_work.genre.add(genre)
+            Trope.objects.filter(origin_id=work_to_redirect.id).update(origin_id=kept_id)
+            models_to_update = [WorkTitle, TaggedWork, Suggestion, Recommendation, Pairing, Reference, ColdStartRating]
+            for model in models_to_update:
+                model.objects.filter(work_id=work_to_redirect.id).update(work_id=kept_id)
+            work_to_redirect.redirect = kept_work
+            work_to_redirect.save()
+        kept_work.save()
+        cluster.resulting_work = kept_work
+        cluster.merged_on = datetime.now()
+        cluster.status = 'accepted'
+        cluster.save()
+        return None
+
+    do_not_compare = ['sum_ratings', 'nb_ratings', 'nb_likes', 'nb_dislikes', 'controversy']
+    priority = {
```

Comment:
It would probably make sense to store those directly on the MergeType enum.

```python
class MergeType(Enum):
    INFO_ONLY = (0, 'black')
    JUST_CONFIRM = (1, 'green')
    CHOICE_REQUIRED = (2, 'red')

    def __init__(self, priority, row_color):
        self.priority = priority
        self.row_color = row_color
```